functional/FunctionalTest.java

```diff
@@ -3761,6 +3764,309 @@ public static void getDefaultRetention_test() throws Exception {
+      } else {
+        mintFailedLog(
+            "getObjectRetention(GetObjectRetentionArgs args)",
+            null,
+            startTime,
+            null,
+            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
+    }
+  }
+
+  /** Test: getObjectRetention(GetObjectRetentionArgs args). */
+  public static void getObjectRetention_test2() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: with shortened retention period: getObjectRetention(GetObjectRetentionArgs args)");
+    }
+
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(MakeBucketArgs.builder().bucket(bucketName).objectLock(true).build());
+      try {
+        try (final InputStream is = new ContentInputStream(1 * KB)) {
+          PutObjectOptions options = new PutObjectOptions(1 * KB, -1);
+          options.setContentType(customContentType);
+          client.putObject(bucketName, objectName, is, options);
+        }
+
+        ZonedDateTime retentionUntil = ZonedDateTime.now().plusDays(3);
+        Retention expectedConfig = new Retention(RetentionMode.GOVERNANCE, retentionUntil);
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(expectedConfig)
+                .build());
+
+        Retention config =
+            client.getObjectRetention(
+                GetObjectRetentionArgs.builder().bucket(bucketName).object(objectName).build());
+
+        if ((!(config
+                .retainUntilDate()
+                .toString()
+                .substring(0, config.retainUntilDate().toString().indexOf("T"))
+                .equals(
+                    expectedConfig
+                        .retainUntilDate()
+                        .toString()
+                        .substring(0, expectedConfig.retainUntilDate().toString().indexOf("T")))))
+            || (config.mode() != expectedConfig.mode())) {
+          throw new Exception(
+              "[FAILED] Expected: expected duration : "
+                  + expectedConfig.retainUntilDate()
+                  + ", got: "
+                  + config.retainUntilDate()
+                  + " expected mode :"
+                  + expectedConfig.mode()
+                  + ", got: "
+                  + config.mode());
+        }
+
+        // You can perform operations on object versions that are locked in governance mode as if
+        // they were unprotected if you have the s3:bypassGovernanceMode permission. These
+        // operations include deleting an object version, shortening the retention period, or
+        // removing the Object Lock by placing a new lock with empty parameters.
+        ZonedDateTime shortenedRetentionUntil = ZonedDateTime.now().plusDays(1);
```

Comment:
1. I think you are testing `bypassGovernanceMode()` here. If so it needs to be merged in above. 
2. I guess `bypassGovernanceMode` allow to change `Retention` where `RetentionUntil` time could be higher or lower.  If this is the test you want to do, you could have the test doing both.

Reply:
This test is to set retention with a shortened retention period. As per s3 , retention period can only be shortened, it can not be extended.
`
You can perform operations on object versions that are locked in governance mode as if they were unprotected if you have the s3:BypassGovernanceRetention permission. These operations include deleting an object version, shortening the retention period, or removing the Object Lock by placing a new lock with empty parameters. 
`